### PR TITLE
Fix bug with missing 'Call reception message'

### LINKED ISSUE #60: No link to GP booking system or 'Phone Reception' message being displayed for some surgeries
When a surgery has a defined booking system supplier for which the link is unknown (see https://github.com/nhsuk/profiles-db/blob/master/lib/bookingSystem.js for the _unknowns_) **and** the surgery does not have a website then the 'Phone Reception' message is not displayed.

An example of this is both branches returned by the query ['Sabden'](https://staging.c2s.nhschoices.net/book-a-gp-appointment/results?search=sabden)

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
Fixes #60

See issue for example search.

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
Fixes #63 

See issue for example search.

File: test/integration/resultsPage.js [hunk 1]
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
...
+        .end((err, res) => {
+          const $ = cheerio.load(res.text);
 
-        done();
+          const resultsHeader = $('.results__header').text();
+          expect(resultsHeader).to.contain(`GP surgeries matching '${search}'`);
+          done();
+        });
+    });
+    describe('matching surgeries found', () => {
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Extra line please

File: test/integration/resultsPage.js [hunk 2]
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
...
+              const $ = cheerio.load(res.text);
+
+              const searchResults = $('.results__item--nearby');
+              expect(searchResults.length).to.equal(1);
+
+              done();
+            });
+        });
+        it('should display singular message text', (done) => {
+          chai.request(app)
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Can this request and the one above be incorporated into one and moved into the `describe` block so the validation of the contents of cheerio can be tested within the `it` blocks. This should reduce duplication and ease readability.

File: test/integration/resultsPage.js [hunk 3]
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
...
-
-        const searchResults = $('.results__item--nearby .results__details').first();
-        expect(searchResults.html()).to.not.contain(noOnlineBookingLinkMessage);
-        expect(searchResults.html()).to.contain('href');
-
-        expect($('.link-back:first-of-type').eq(0).text()).to.equal('Back');
-        expect($('.link-back:first-of-type').eq(1).text()).to.equal('Back');
-        expect($('.link-back').attr('href')).to.equal(`${constants.SITE_ROOT}`);
-        done();
+      describe('multiple matches', () => {
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Extra line please

File: test/integration/resultsPage.js [hunk 4]
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
...
+
+              const searchResults = $('.results__item--nearby');
+              expect(searchResults.length).to.be.above(1);
+
+              done();
+            });
+        });
+        it('should display plural message text', (done) => {
+          const search = 'Surgery';
+          chai.request(app)
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Same as previous comment about removing duplication and running the request within the `describe` block

File: test/integration/resultsPage.js [hunk 5]
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
...
+
+            const searchResults = $('.results__item--nearby .results__details .results__name a').first();
+            expect(searchResults.text()).to.equal('Hambleden Surgery');
+            expect(searchResults.attr('href')).to.equal('http://www.marlowdoctors.co.uk');
+
+            done();
+          });
+      });
+    });
+    describe('when the surgery does not have a website', () => {
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Extra line please

File: test/integration/resultsPage.js [hunk 6]
```diff
@@ -169,8 +248,8 @@ describe('Results page error handling', () => {
     });
   });
   describe('when search is an empty string', () => {
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Extra line please

File: test/integration/resultsPage.js [hunk 7]
```diff
@@ -186,31 +265,22 @@ describe('Results page error handling', () => {
...
-          const errorHeader = $('#content').text();
-          expect(errorHeader).to.contain(errorMessage);
+          expect($('.link-back:first-of-type').length).to.equal(0);
           done();
         });
     });
   });
-
-  describe("when search doesn't bring back results", () => {
+  describe('when search is some empty spaces', () => {
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Extra line please

File: test/integration/resultsPage.js [hunk 8]
```diff
@@ -15,136 +15,216 @@ const resultsRoute = `${constants.SITE_ROOT}/results/`;
...
+
+            const searchResults = $('.results__item--nearby .results__details').first();
+            expect($('.callout p', searchResults).text().trim()).to.equal('This surgery doesn\'t have an online booking system. Call reception on 0114 236 0641 to book an appointment.');
+            expect($('a[href^="tel:"]', searchResults).text()).to.equal('0114 236 0641');
+
+            done();
+          });
       });
+    });
+    describe('Surgeries without phone number', () => {
```
Comment (st3v3nhunt) [inline test/integration/resultsPage.js]: Extra line please

**Comment by neilbmclaughlin** [comment]
@st3v3nhunt - will refactor the tests to incorporate your suggestions.

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
@st3v3nhunt - I have addressed some of the issues you raise but would like to address the issue of refactoring the tests to use Promises, beforeEach and describe block better as part of a separate pull request so the bug fix here can be released. The reason being was that I was getting bogged down with issues with Docker volumes. Raised issue #66 for the specific problem of too much logic in the views.

Are you OK the merge on the basis of this?

File: test/integration/resultsPageErrors.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+const iExpect = require('../lib/expectations');
+const messages = require('../../app/lib/messages');
+
+const expect = chai.expect;
+
+chai.use(chaiHttp);
+
+const resultsRoute = `${constants.SITE_ROOT}/results/`;
+
+function assertSearchResponse(search, assertions) {
```
Comment (neilbmclaughlin) [inline test/integration/resultsPageErrors.js]: I realise there is duplication here with the resultsPage tests but it is just an interim step before we use beforeEach.

**Comment by st3v3nhunt** [comment]
@neilbmclaughlin There is still a bunch of inconsistent spacing within the tests either no new line where there should be or an additional new line where there shouldn't be. Will this also be dealt with in the next change? Ideally it would be good to get it done as part of this

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
@st3v3nhunt - experimented with a test specific lint rule around padding. This may not be quite what you want but hopefully is close enough. Importantly it allows us to pick issues up automatically rather than at the review stage and allows us to fix them automatically using `npm run lint -- --fix`. The is perhaps a discussion to be had as to whether the test specific padding rule should be applied to the whole of the code base since `--fix` makes it a trivial task.

**Comment by st3v3nhunt** [comment]
@neilbmclaughlin I think that change makes it harder to read. The previous set of comments about where to add lines and remove them covered all of the suggestions.
Which equate to this...the general rule I work with, and is implicitly used by other people across the set of `connecting-to-service` associated repos is have no padding for blocks with the exception of between `it` blocks. The reasoning for padding between `it` blocks is to aid readability and have each individual test be visibly separate from one another.
Looking at the eslint rule for [padded-blocks](http://eslint.org/docs/rules/padded-blocks) there isn't the configurability to specify a rule based on mocha flavoured blocks.

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
@st3v3nhunt - I've overridden the no-padding rule so we can pad as we wish in tests and `git push --force`'d a reformatted set of tests. This will allow us to visually separate test blocks without adding unnecessary white space.  My frustration with this was that the repo eslint rule seemed to be at odds with what you think we should be doing - rules have make coherent sense to me in order to want to do them. I think if we want to preceded a test block with a blank line (and I think this is a good idea) it should apply regardless of how many tests there are and whether it is an 'it' or a 'describe'. Hopefully we are converging :)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-63.dev.c2s.nhschoices.net](http://gp-finder-pr-63.dev.c2s.nhschoices.net)

**Pull request merged**